### ISSUE TITLE
DataLoader for different input shapes

### ISSUE THREAD

**Issue by TJPaik** [pytorch/pytorch#57351]
## 🚀 Feature

DataLoader for different input shapes using custom batch sampler!

## Motivation and Detail

I used pytorch to predict some diseases from ecg data. (medical, 1 dimentional time series data)
Different hospitals have different lengths of ecg data because of various ecg recording machines. Instead of cropping or padding, I decided to use intact waves while doing a project.
So I used custom batch sampler to deal with this kind of data. I think this can be applied a lot in other data structure, so I made this 'issue'. What do you guys think?

## ETC
I'm new to this github. Is this right way to do this? :open_mouth: 

cc @cpuhrsch

**Comment by TJPaik** [comment]
Thank you! :smile:

**Comment by cpuhrsch** [comment]
Hello @TJPaik - The nestedtensor project might be useful to you (actual binaries compatible with the latest nightly will be available soon!). See the whirlwind tutorial and please don't hesitate to open an issue to ask for op coverage or other usecases.
